### PR TITLE
Support custom IPFS gateway URLs

### LINKED ISSUE #145: Support for custom IPFS gateways
* **Motivation:** 
I was setting up a new IPFS server today. Usually, I prefer installing the latest binary through `ipfs-update`, because it is easy to patch new updates and migrate fs repos. For the past few days, I have experienced some disruption with the main IPFS gateway. Here's the console output:
```sh
$ ipfs-update install latest
ERROR: error resolving "latest": GET https://ipfs.io/ipns/dist.ipfs.io/go-ipfs/versions error: 429 Too Many Requests: <html>
<head><title>429 Too Many Requests</title></head>
<body>
<center><h1>429 Too Many Requests</h1></center>
<hr><center>openresty</center>
</body>
</html>
```

After digging around a little bit, I discovered a similar issue was filed some time back: #105. We can remove this sole dependency on ipfs.io gateway and allow users to specify any custom IPFS gateway of choice.

  * **How the feature should work:** 
     * As a user, when I expose my own custom IPFS gateway URL of choice through an OS environment variable,
the `ipfs-update` should use this URL to query for versions and fetch the binary.  
For example: `$ IPFS_CUSTOM_GATEWAY_URL="https://dweb.link" ipfs-update install latest`  
     * As a user, when I do not expose a custom IPFS gateway URL, then `https://ipfs.io` will be used as the default gateway by `ipfs-update`.

### PR DESCRIPTION
**Pull request by 0zAND1z**
Fixes #145 
 - Code changes to read the custom gateway URLs by OS environment variables
 - Update README to guide users to export and use custom gateway URLs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +104,16 @@ algorithm:
...
 [go-env]: https://golang.org/cmd/go/#hdr-Environment_variables
 
+## Custom IPFS gateway URL
+
+By default, `ipfs-update` uses https://ipfs.io as the gateway URL. If you wish to use your own IPFS gateway URL, please export it via the environment variable `IPFS_CUSTOM_GATEWAY_URL`.
+
+For example:
+
+```sh
+$ IPFS_CUSTOM_GATEWAY_URL="https://dweb.link" ipfs-update install latest
```
Comment (Stebalien) [inline README.md:114]: nit: we usually use `IPFS_GATEWAY` (at least that's what I have set in my local environment).

File: main.go [hunk 2]
```diff
@@ -314,6 +314,12 @@ func createFetcher(c *cli.Context) migrations.Fetcher {
...
 
+	if customIpfsGatewayURL := os.Getenv("IPFS_CUSTOM_GATEWAY_URL"); customIpfsGatewayURL != "" {
+		return migrations.NewMultiFetcher(
+			lib.NewIpfsFetcher(distPath, 0),
+			migrations.NewHttpFetcher(distPath, customIpfsGatewayURL, userAgent, 0))
+	}
+
 	return migrations.NewMultiFetcher(
 		lib.NewIpfsFetcher(distPath, 0),
 		migrations.NewHttpFetcher(distPath, "", userAgent, 0))
```
Comment (Stebalien) [inline main.go:325]:
```suggestion
	customIpfsGatewayURL := os.Getenv("IPFS_GATEWAY") // uses the default if unset

	return migrations.NewMultiFetcher(
		lib.NewIpfsFetcher(distPath, 0),
		migrations.NewHttpFetcher(distPath, customIpfsGatewayURL, userAgent, 0))
```

**Review by Stebalien — CHANGES_REQUESTED** [review]
Two small nits, otherwise lgtm.

**Comment by Stebalien** [comment]
And thanks!

File: README.md (hunk 1 above)
Comment (0zAND1z) [reply README.md]: I have updated the README. Thanks for throwing some light on this, didn't realize that a env. var. was already under use.

File: main.go (hunk 2 above)
Comment (0zAND1z) [reply main.go]: I have made the suggested changes.

**Review by gammazero — APPROVED** [review]

**Review by Stebalien — APPROVED** [review]

**Pull request merged**